LibreHardwareMonitorLib/Hardware/SMBios.cs

```diff
@@ -648,12 +648,23 @@ internal BiosInformation(byte[] data, IList<string> strings) : base(data, string
             int.TryParse(parts[1], out int day) &&
             int.TryParse(parts[2], out int year))
         {
+            if (day > 31)
+            {
+                (year, day) = (day, year);
+            }
+
             if (month > 12)
             {
                 (month, day) = (day, month);
             }
 
-            return new DateTime(year < 100 ? 1900 + year : year, month, day);
+            try {
+                return new DateTime(year < 100 ? 1900 + year : year, month, day);
+            }
```

Comment:
This isn't formatted properly, but is try/catch still needed now? Days and months should now no longer be above their allowed values?

Reply:
Because I think it still can't check the format, is "yyyy/mm/dd" or "mm/dd/yyyy" or "dd/mm/yyyy" or others?
And once convert the value of datetime is wrong, the program will throw an exception.

Reply:
Of course, if u think it is OK, u can add the check “yyyy/mm/dd” only, thanks.

Reply:
Of course, if u think it is OK, u can add the check “yyyy/mm/dd” only, thanks.

Reply:
DateTime ctor is year, month and day, which should be checked with the lines above it. I don't have a BIOS that violates the specification so I'm not entire sure what the format is, but as long as days are <= 31 and months are <= 12 it should not throw anymore.

Reply:
No need to close, the day check is still useful.

Reply:
Yes, I repulled the project

Reply:
I'm sorry, I'm not familiar with the process.